truss/contexts/image_builder/serving_image_builder.py

```diff
@@ -144,3 +157,82 @@ def _render_dockerfile(
         )
         docker_file_path = build_dir / MODEL_DOCKERFILE_NAME
         docker_file_path.write_text(dockerfile_contents)
+
+    def _download_external_data(self, data_dir: Path):
+        external_data = self._spec.external_data
+        if external_data is None:
+            return
+        data_dir.mkdir(exist_ok=True)
+        b10cp_path = _b10cp_path()
+
+        # ensure parent directories exist
+        for item in external_data.items:
+            path = data_dir / item.local_data_path
+            if data_dir not in path.parents:
+                raise ValueError(
+                    "Local data path of external data cannot point to outside data directory"
+                )
+            path.parent.mkdir(exist_ok=True)
+
+        if b10cp_path is not None:
+            print("b10cp found, using it to download external data")
+            _download_external_data_using_b10cp(b10cp_path, data_dir, external_data)
+            return
+
+        # slow path
+        _download_external_data_using_requests(data_dir, external_data)
+
+
+def _b10cp_path() -> Optional[str]:
+    return os.environ.get(B10CP_PATH_TRUSS_ENV_VAR_NAME)
+
+
+def _download_external_data_using_b10cp(
+    b10cp_path: str,
+    data_dir: Path,
+    external_data: ExternalData,
+):
+    procs = []
+    for item in external_data.items:
```

Comment:
Is there any downside to not limiting the concurrency here?

Reply:
Good point. Not currently, we don't have external data used extensively right now. But let me leave a todo here.